src/groups/bmq/bmqp/bmqp_putmessageiterator.cpp

```diff
@@ -607,12 +615,10 @@ int PutMessageIterator::next()
         // (with lengths, not offsets).  That style does not work with schema.
         // So, leave the schema absence. The only effect is de-compressing.
         decompressFlag = true;
-        d_header.setCompressionAlgorithmType(
-            bmqt::CompressionAlgorithmType::e_NONE);
-        // No need to write the header back to the blob since future events
-        // will use 'header()' as their input (not the blob).
```

Comment:
@dorjesinpo I moved this call to the bottom.
Before this PR, we had a following sequence of actions for decompressing old message properties:
`
save algorithm type to 'cat'
set decompress flag
set algorithm type to NONE
decompress using 'cat'
recalculate crc32
`

I changed the order to:
`
set decompress flag
save algorithm type to 'cat'
decompress using 'cat'
set algorithm type to NONE
recalculate crc32
`

Reply:
Yeah, we need to read `cat` before  `d_header.setCompressionAlgorithmType(bmqt::CompressionAlgorithmType::e_NONE);`